templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -0,0 +1,313 @@
+mode: 0755
+path: "/usr/local/bin/configure-ovs.sh"
+contents:
+  inline: |
+    #!/bin/bash
+    set -eux
+
+    # Configures NICs onto OVS bridge "br-ex"
+    # Configuration is either auto-detected or provided through a config file written to
+    # /etc/sysconfig/network-scripts/ovs-config.yaml
+    # Example Config file:
+    # ---
+    # interfaces:
+    #   - name: br-ex
+    #     type: ovs-bridge
+    #     state: up
+    #     bridge:
+    #       port:
+    #         - name: bond0
+    #           vlan:
+    #             mode: access
+    #             tag: 2
+    #           link-aggregation:
+    #             lacp: active
+    #             mode: balance-slb
+    #             slaves:
+    #               - name: eth1
+    #               - name: eth2
+    #         - name: br-ex
+    #   - name: br-ex
+    #     type: ovs-interface
+    #     mtu: 9000
+    #     mac-address: 52:54:00:E0:64:7D
+    #
+    # This configuration file format is the same as what is used by nmstate. Only configuration options listed
+    # here are supported. The only configuration this file should contain is for configuring br-ex. Other interfaces
+    # are not supported.
+
+    parse_yaml() {
+      local yaml_file=$1
+      local prefix=$2
+      local s
+      local w
+      local fs
+
+      s='[[:space:]]*'
+      w='[a-zA-Z0-9_.-]*'
+      fs="$(echo @|tr @ '\034')"
+
+      (
+        sed -e '/- [^\“]'"[^\']"'.*: /s|\([ ]*\)- \([[:space:]]*\)|\1-\'$'\n''  \1\2|g' |
```

Comment:
Use `$s` instead of `[[:space:]]*`

Reply:
Also, can you add some comments on what this sed/awk black-magic does so someone trying to grok this later can have an easier time?

Reply:
It's magic to me as well :) I've used this in the past on another project for yaml parsing in bash, so it works. The hope is this is short-lived and replaced by something in golang when we can figure out how to do that with MCD. Or if we could get yq installed we could get rid of these functions if we stick with bash.